### ISSUE TITLE
Visualizing multiple different modifications

### ISSUE THREAD

**Issue by gevro** [igvteam/igv#1425]
Hi,
Is there a possibility of IGV adding the ability to visualize multiple different modifications (ML / MM tags) simultaneously? For example, each MOD in a different color, with a legend.

SAM specs allows for multiple different mods, and there are technologies that can produce this. So I think it would be really good to have this.

Thanks

**Comment by jrobinso** [comment]
Anything is possible of course, but unfortunately I am not made of time.   I would entertain pull requests for this enhancement if it was minimally disruptive to current code.  For the (very) small IGV team  "there are technologies that can produce this" is not a high enough threshold to prioritize, there needs to be a significant user community actually using it.

**Comment by gevro** [comment]
Thanks. Understood. How do you all generally assess how many in the community may use a potential feature? Is there a survey sent periodically to users to which this feature could be posed? It is hard for me to know how many may need this. It would be nice if github itself had a feature voting page!

I do not code well enough to attempt this, but I think that with Nanopore and Pacbio's recent advances in sequencing multiple modifications (which the SAM spec anticipated) that this will increasingly occur.

**Comment by jrobinso** [comment]
We cannot do surveys, we don't even have contact info,  but generally we are pretty plugged into the community.   Generally we rely on guidance from Oxford Nanopore and Pacbio development teams, requests here and in the igv-help forum, and citations of IGV for particular features.   The entire IGV development team here is ~ 1 person, so we have to priortize.

**Comment by gevro** [comment]
Ok. Well, I assume this will be considered as one vote for this feature, and if other users agree, perhaps they can comment here on this thread. I'm happy to help in other ways and wish I knew how to code this to be able to help - sorry!

**Comment by MBoemo** [comment]
Related to this:
Thanks very much to @jrobinso and the devs for all of the great work on incorporating base modifications. A method that's quite important to the genome replication community is labelling nascent DNA with thymidine analogues BrdU and EdU. We and others have done quite a bit of work using the ONT platform for this (https://www.biorxiv.org/content/10.1101/2022.09.22.509021v1) and a new version of our tool (https://github.com/MBoemo/DNAscent) will output BrdU and EdU calls as modbam. We could, in theory, visualise these by pretending they're one of the other modified bases that IGV currently supports but I think it would be better to do this properly. There's no single-letter code defined for these analogues, but we could use ChEBI IDs. Would you be happy to consider a pull request if I put these in?

**Comment by jrobinso** [comment]
It should (nearly) just work, have you tried?   If not could you give me a small example file.   There are probably a few places where there is some hardcoded assumption its a known modification.

**Comment by gevro** [comment]
Hi all, Just to add, the original suggestion was to visualize multiple modifications simultaneously (with a legend). That will become an increasingly common situation I think, as multiple modifications are called by PacBio and nanopore sequencing.

**Comment by MBoemo** [comment]
Everything works as expected if, for instance, we put "N+b?" and "N+e?" in the MM tag for BrdU and EdU, respectively. Then we can pick colours via Preferences --> Base Mods --> Modified base colours and visualise them that way. The slightly sketchy thing about that is that IGV then calls them 5fU and 5caU which isn't correct.

I took a quick look earlier and I think (hopefully not famous last words) that this could be implemented by adding only a few lines of code.

**Comment by MBoemo** [comment]
Yes I agree @gevro - we could probably put two new modified bases in quite easily, but then we'll have to go through this again every time there's a new one in the future.

**Comment by jrobinso** [comment]
@MBoemo I also looked briefly and yes, it needs a few lines of code.   A PR would be welcome, along with an example file for testing.
